lib/ansible/modules/system/user.py

```diff
@@ -511,6 +511,9 @@ def __init__(self, module):
         else:
             self.ssh_file = os.path.join('.ssh', 'id_%s' % self.ssh_type)
 
+        if self.groups is None and self.append:
+            module.warn("Append is set, but no groups are specified. Use 'groups' for appending new groups.")
```

Comment:
```suggestion
            # Change the argument_spec in 2.14 and remove this warning
            # required_by={'append': ['groups']}
            module.warn("'append' is set, but no 'groups' are specified. Use 'groups' for appending new groups."
                        "This will change to an error in Ansible 2.14.")
```